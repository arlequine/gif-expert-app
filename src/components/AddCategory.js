import React, { useState } from 'react'
import PropTypes from 'prop-types';


const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = e => {
    console.log(e.target.value)
    setInputValue(e.target.value)
  }
  const handleSubmit = e => {
    console.log('->handleSubmit', inputValue, '<--')
    e.preventDefault()
    console.log(inputValue)
    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }

  }
  return (
    <>
      <h2>Add Category</h2>
      <p>{inputValue}</p>
      <form onSubmit={handleSubmit}>
        <input
          type='text'
          value={inputValue}
          onChange={handleInputChange}
          placeholder='Category Name'
        />
      </form>
    </>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;