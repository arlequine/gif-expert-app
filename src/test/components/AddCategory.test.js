import React from 'react';
import AddCategory from '../../components/AddCategory';
import { shallow } from 'enzyme';


describe('Pruebas en componenten <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper

  beforeEach(() => {
    jest.clearAllMocks(); //si tenemos alguna simulacion previa, todo se limpia 
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });


  test('Deberia tener el formulario correcto', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('deberia debe de cambiar la caja de texto', () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value: value } });
    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('no debe de mandar la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() { } });//forma corta de la funcion preventDefault
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('debe de llamar el setCategories y limpiar la caja de texto', () => {
    const val = 'One Piece';
    const input = wrapper.find('input');


    // 1 simular inputChange
    input.simulate('change', { target: { value: val } });

    // 2 simular submit
    wrapper.find('form').simulate('submit', { preventDefault() { } });//forma corta de la funcion preventDefault

    // 3 verificar que se llame al setCategories
    expect(setCategories).toHaveBeenCalled();
    // expect(setCategories).toHaveBeenCalledTimes(2); 
    // .toHaveBeenCalledTimes(2) se manda a llamar la funcion segun el valor del argumento se le ponga
    // expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
    // se espera que sea llamado con cualquier tipo de funcion 

    // 4 verificar que el value del input sea vacio
    expect(input.prop('value')).toBe('');
    // al colocar prop se hace referencia a las propiedades del input

  });

});
