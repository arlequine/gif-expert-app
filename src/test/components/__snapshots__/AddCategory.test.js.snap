// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componenten <AddCategory /> Deberia tener el formulario correcto 1`] = `
<Fragment>
  <h2>
    Add Category
  </h2>
  <p />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Category Name"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
